storage_controller/src/service.rs

```diff
@@ -3764,10 +3811,169 @@ impl Service {
 
         Ok(TimelineCreateResponseStorcon {
             timeline_info,
-            safekeepers,
+            safekeepers: selected_safekeepers,
         })
     }
 
+    pub(crate) async fn put_timeline_import_status(
```

Comment:
nit: `struct Service` usually don't have HTTP verbs in their name

Reply:
`handle_timeline_shard_import_progress_upcall` ?

Reply:
Done in https://github.com/neondatabase/neon/pull/11345/commits/29d1872256b6e9e05167b01191cf14f757a206e9